### PR TITLE
fix: Pinpoint session generation

### PR DESCRIPTION
**Pull request by jimblanc**
#### Description of changes

Fixes session generation for Pinpoint when auto-track events are enabled. Validated behavior against v5 session management.

#### Description of how you validated changes
Tested in sample apps.

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#steps-towards-contributions)
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/core/__tests__/providers/pinpoint/apis/record.test.ts [hunk 1]
```diff
@@ -111,6 +111,63 @@ describe('Pinpoint Provider API: record', () => {
 		);
 	});
 
+	it('should start and stop sessions when appropriate events are received', async () => {
+		const mockStartEvent = {
+			name: '_session.start',
```
Comment (cshfang) [inline packages/core/__tests__/providers/pinpoint/apis/record.test.ts:116]: Nit: Get from constants?

**Review by cshfang — APPROVED** [review]

File: packages/core/src/providers/pinpoint/apis/record.ts [hunk 2]
```diff
@@ -64,6 +68,16 @@ export const record = async ({
...
 
+	// Terminate session when required
+	if (session && event.name === SESSION_STOP_EVENT) {
+		session = {
+			...session,
+			StopTimestamp: timestampISOString,
+			Duration:
+				currentTime.getTime() - new Date(session.StartTimestamp).getTime(),
+		};
+	}
```
Comment (elorzafe) [inline packages/core/src/providers/pinpoint/apis/record.ts:79]: should `session` be cleaned after is recorded? for example session stop is called twice?

Comment (jimblanc) [reply packages/core/src/providers/pinpoint/apis/record.ts:79]: Good call, that is probably cleaner. It'll regenerate a new session on the next `record` if the previous one was stopped.

**Review by cshfang — APPROVED** [review]

**Review by elorzafe — APPROVED** [review]
Looks good!

Thanks @jimblanc 🎖️

**Pull request merged**